golos.publication/golos.publication.cpp

```diff
@@ -435,15 +442,64 @@ void publication::close_message(structures::mssgid message_id) {
             eosio_assert(state.funds.amount >= 0, "LOGIC ERROR! publication::payrewards: state.funds < 0");
         }
 
-        auto new_rshares = WP(state.rshares) - wdfp_t(FP(mssg_itr->state.netshares));
-        auto new_rsharesfn = WP(state.rsharesfn) - wdfp_t(sharesfn);
+        auto rshares = wdfp_t(FP(mssg_itr->state.netshares));
+        auto new_rshares = WP(state.rshares) - rshares;
+
+        auto rsharesfn = wdfp_t(sharesfn);
+        auto new_rsharesfn = WP(state.rsharesfn) - rsharesfn;
 
         eosio_assert(new_rshares >= 0, "LOGIC ERROR! publication::payrewards: new_rshares < 0");
         eosio_assert(new_rsharesfn >= 0, "LOGIC ERROR! publication::payrewards: new_rsharesfn < 0");
 
         state.rshares = new_rshares.data();
         state.rsharesfn = new_rsharesfn.data();
+        
+        mssg_reward.rshares = rshares.data();
+        mssg_reward.rsharesfn = rsharesfn.data();
```

Comment:
As I can see `rshares` and `rsharesfn` doesn't used in reward calculation in `paymssgrwrd`. So, you doesn't need to save it.